tests/test_utils.py

```diff
@@ -119,26 +128,28 @@ def test_url_handler(url_type, url):
         assert is_opendap_url(processed)
 
 
+@pytest.mark.parametrize(
+    ("netcdfs", "argc"),
+    [([local_path("tiny_daily_pr.nc"), local_path("tiny_daily_prsn.nc"),], 3,)],
+)
```

Comment:
I would like to see more complex test cases. `collect_args` can collect multiple arguments per input, and that is why I want `TestCollectArgs` to have multiple `ComplexInput`s. For example, try 3 arguments for file1 and 2 arguments for file2. See if `collect_args` collects the arguments paired with the right key (identifier) rather than `assert`ing the total number of arguments.